pkg/cloudclient/aws/private.go

```diff
@@ -868,8 +868,20 @@ func (ac *Client) removeLoadBalancerFromMasterNodes(ctx context.Context, kclient
 	var intDNSName, intHostedZoneID, lbName string
 	for _, networkLoadBalancer := range nlbs {
 		if networkLoadBalancer.scheme == "internet-facing" {
+
+			var nlbsToDeleteArn string
+			nlbTagsMap, err := ac.getAllTagsFromLoadBalancer(networkLoadBalancer.loadBalancerArn)
+			if err != nil {
+				fmt.Printf("There was a problem getting back network load balancer tags for NLB %s: %s\n", networkLoadBalancer.loadBalancerArn, err)
+				return "", "", err
+			}
+
 			lbName = networkLoadBalancer.loadBalancerName
-			err := ac.deleteExternalLoadBalancer(networkLoadBalancer.loadBalancerArn)
+			if _, ok := nlbTagsMap["kubernetes.io/service-name"]; !ok {
+				nlbsToDeleteArn = networkLoadBalancer.loadBalancerArn
+			}
+
+			err = ac.deleteExternalLoadBalancer(nlbsToDeleteArn)
```

Comment:
This can be called with an empty String if you never hit your `if` - not sure what will happen in that case - did you verify that?

Might be better to handle the empty string case directly before relying on AWS to do what we hope it does.

Reply:
Done